File: pytools/data/data.code.lua53.client/ReqSceneEnterFly.lua
ReqSceneEnterFly = {}
setmetatable(ReqSceneEnterFly, {__index = _G})
local _ENV = ReqSceneEnterFly


function new()
	local tab_obj = {}
	setmetatable(tab_obj, {__index = ReqSceneEnterFly})
	tab_obj:init()
	return tab_obj
end

function init(self)
	self.map_id = 0
end


function encode(self)
	pack = packet.new()
	pack:writeU32(self.map_id)
	return pack:encode(Msg.P_REQ_SCENE_ENTER_FLY)
end


function setMapId(self, map_id)
	self.map_id = map_id
end

File: pytools/data/data.code.golang.client/req_scene_enter_fly.go
package proto

import (
	"packet"
)

type ReqSceneEnterFly struct {
	mapId                    uint32
}

func (this *ReqSceneEnterFly) Encode() []byte {
	pack := packet.NewWriteBuff(64)

	pack.WriteUint32(this.mapId)

	return pack.Encode(P_REQ_SCENE_ENTER_FLY)
}

func (this *ReqSceneEnterFly) SetMapId(mapId uint32) {
	this.mapId = mapId
}
